org.adempiere.base/src/org/compiere/model/MBankStatementLine.java

```diff
@@ -269,5 +278,19 @@ protected boolean updateHeader()
 		}
 		return true;
 	}	//	updateHeader
+
+
+	/**
+	 * Test if the line's date is on the same period as header's, only if BANK_STATEMENT_POST_WITH_DATE_FROM_LINE SysConfig is set to Y 
+	 */
+	public boolean checkDateIfUsedForPosting() {
+		if (MBankStatement.isPostWithDateFromLine(getAD_Client_ID())) {
+			MPeriod headerPeriod = MPeriod.get(getCtx(), getParent().getDateAcct(), getParent().getAD_Org_ID(), get_TrxName());
+			MPeriod linePeriod = MPeriod.get(getCtx(), getDateAcct(), getParent().getAD_Org_ID(), get_TrxName());
+
+			return headerPeriod != null && linePeriod != null && headerPeriod == linePeriod;	

```

Comment:
should use ID for equal comparison instead.

Reply:
done